FlowCrypt/Controllers/Setup/SetupEKMKeyViewController.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  SetupEKMKeyViewController.swift
+//  FlowCrypt
+//
+//  Created by Yevhen Kyivskyi on 13.08.2021.
+//  Copyright © 2021 FlowCrypt Limited. All rights reserved.
+//
+
+import AsyncDisplayKit
+import FlowCryptCommon
+import FlowCryptUI
+import Promises
+
+enum CreatePassphraseWithExistingKeyError: Error {
+    // No private key was found
+    case noPrivateKey
+}
+
+/**
+ * Controller which is responsible for setting up a keys received from EKM
+ * - User is sent here from **SetupInitialViewController** in case he has keys on EKM
+ * - Here user can enter a pass phrase (will be saved in memory)
+ * - After passphrase is set up, user will be redirected to **main flow** (inbox view)
+ */
+
+final class SetupEKMKeyViewController: SetupCreatePassphraseAbstractViewController {
+
+    override var parts: [SetupCreatePassphraseAbstractViewController.Parts] {
+        SetupCreatePassphraseAbstractViewController.Parts.ekmKeysSetup
+    }
+    private let keys: [CoreRes.ParseKeys]
+
+    private lazy var logger = Logger.nested(in: Self.self, with: .setup)
+
+    init(
+        user: UserId,
+        keys: [CoreRes.ParseKeys] = [],
+        core: Core = .shared,
+        router: GlobalRouterType = GlobalRouter(),
+        decorator: SetupViewDecorator = SetupViewDecorator(),
+        storage: DataServiceType = DataService.shared,
+        keyStorage: KeyStorageType = KeyDataStorage(),
+        passPhraseService: PassPhraseServiceType = PassPhraseService()
+    ) {
+        self.keys = keys
+        super.init(
+            user: user,
+            core: core,
+            router: router,
+            decorator: decorator,
+            storage: storage,
+            keyStorage: keyStorage,
+            passPhraseService: passPhraseService
+        )
+        self.shouldStorePassPhrase = true
```

Comment:
You have to read issue descriptions more carefully, I've spent a lot of effort to outline the issue in great detail, and it's wasted effort if it's not read and followed.

Have a look at this https://user-images.githubusercontent.com/6306961/124514325-4f64b900-ddcc-11eb-9419-01973f122a37.png
and this https://fes.flowcrypt.com/api/v1/client-configuration?domain=flowcrypt.com
And this https://github.com/FlowCrypt/flowcrypt-ios/issues/279
And this https://flowcrypt.com/docs/business/org-rules.html

You should see a discrepancy very easily.

Reply:
oh, I did check all the docs and understand the flow, just messed up with name of variable :(

Reply:
Ok. Thank you for responding - it's really important to be careful.